Lib/Src/CUartCom.cpp

```diff
@@ -133,17 +142,9 @@ bool CUartCom::transmit()
     if (m_tx_queue.size() > 0)
     {
         // If working with half-duplex, disable RX interrupts
-        if (mb_half_duplex)
+        if (mb_half_duplex && m_status == IDLE)
```

Comment:
To avoid calling this function each time transmit is called, I added a condition to the m_status. If the status is TX, the RX interrupts were already disabled.